### ISSUE TITLE
Root permission granting is not properly handled

### ISSUE THREAD

**Issue by aviraxp** [topjohnwu/MagiskManager#149]
Sometimes, when I uninstall an app, it is not removed from the database, so if other apps are installed and use the same identity symbol (UID?), it will be automatically granted root access.

**Comment by aviraxp** [comment]
Seems this will happen when installing application with same package name but different name.

**Comment by Didgeridoohan** [comment]
I saw this today where [Root Checker](https://play.google.com/store/apps/details?id=com.joeykrim.rootcheck) was granted superuser access without prompting for it. The toast stated that [GSAM Battery Monitor](https://play.google.com/store/apps/details?id=com.gsamlabs.bbm) was granted superuser access, an app that I currently do not have on my device (but have granted superuser access previously)...

I did some testing (installing - granting root - uninstalling, etc) to see if I could repeat the issue, but no dice and therefore no logs. :(

Magisk v12, Magisk Manager 4.3.3.
